lib/jnpr/junos/utils/sw.py

```diff
@@ -1037,7 +1040,10 @@ def poweroff(self, in_min=0, on_node=None):
             cmd = E('request-power-off')
             if self._multi_RE is True and self._multi_VC is False:
                 cmd.append(E('both-routing-engines'))
-        cmd.append(E('in', str(in_min)))
+        if in_min >= 0 and at is None:
```

Comment:
Order of preference in JUNOS: `in_min` is honored before `at`.  Can you change the if condition to do the same ??

For reference, logic in Ansible:
https://github.com/rsmekala/ansible-junos-stdlib/blob/9fa1ddc899923dcc224d9cce272ae55d9fc20966/library/juniper_junos_system.py#L377

Reply:
Yes, this patch works as your expects.
* 1st priority: in_min
* 2nd priority: at
Thank you.
## This patch referenced SW.reboot().

Reply:
The expected behavior is not met in the current logic.

On executing the below script:
```
from jnpr.junos.device import Device
from jnpr.junos.utils.sw import SW

with Device('xx.xx.xx.xx', user='xxx', passwd='xxxxx') as dev:
    sw = SW(dev)
    sw.poweroff(in_min=2, at="1903182100")
```

I see the below netconf trace.
```
Mar 18 20:01:59 [NETCONF] - [12475] Incoming: <?xml version="1.0" encoding="UTF-8"?><nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:30b716f5-26e2-4cbb-9cc9-2258f942bdba"><request-power-off><bot
h-routing-engines/><at>1903182100</at></request-power-off></nc:rpc>]]>]]>
```

The current logic checks if `in_min >= 0 and at is None` which is `False` in the above case hence `at` is appended instead of `in_min`

Reply:
Thanks for pointing out `sw.reboot`, will fix it in another PR.

Reply:
Thank you for your review.
I understand my bug. I’ll fix it soon in another PR.